File: Assets/Scripts/inputDriver.js
private var PortChoice : String = "None";

function Start () {
	if (GameObject.Find("ParamHolder")) {
		var paramHolder = GameObject.Find("ParamHolder").GetComponent("ParamHolder");
		PortChoice = paramHolder.portChoice;
	}
	
	if (PortChoice != "None") {
		Destroy(gameObject.GetComponent("inputDriver"));
	}
}

function Update () {
	var thisX : float = 0.0;
	var thisY : float = 0.0;

	thisX = Input.GetAxis("Horizontal");
	thisY = Input.GetAxis("Vertical");
	//Debug.Log("X: "+thisX+" Y: "+thisY);

	this.GetComponent("ShipController").SendMessage("inputMovement", new Vector2(thisX, thisY));
}

File: Assets/Scripts/inputDriver.cs
using UnityEngine;
using System.Collections;

public class inputDriver : MonoBehaviour {

	private string PortChoice = "None";
	private ShipController ship;

	// Use this for initialization
	void Start () {
		if (GameObject.Find("ParamHolder")) {
			ParamHolder paramHolder = GameObject.Find("ParamHolder").GetComponent<ParamHolder>();
			PortChoice = paramHolder.portChoice;
		}
		
		if (PortChoice != "None") {
			Destroy(gameObject.GetComponent<inputDriver>());
		}

		ship = GetComponent<ShipController>();
	}

	void Update () {
		var move = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"));

		ship.velocity = move;

		// Waiting for input?
		if (ship.permImmune && Input.anyKey) {
			StartCoroutine(ship.Immunity(2f));
		}
	}
	
}
